core/model/modx/modparser.class.php

```diff
@@ -31,6 +31,25 @@
  */
 class modParser {
     /**
+     * State 1: Parser ignores uncached tags and does not remove undefined tags.
+     *
+     * @var string
+     */
+    public static $STATE_CACHED = '1_cached';
+    /**
+     * State 2: Parser parses both cached and uncached and does not remove undefined tags.
+     *
+     * @var string
+     */
+    public static $STATE_UNCACHED = '2_uncached';
+    /**
+     * State 3: Parser parses both cached and uncached and removes undefined tags.
+     *
+     * @var string
+     */
+    public static $STATE_REMOVE = '3_remove';
+
+    /**
```

Comment:
All this extra code is for clarity for people who are unfamiliar with the parser. It took me a long time stepping through the code to figure out the MODX parsing logic, and a simple _isGloballyProcessingUncached instance variable that is used in only one place would just add to the confusion. My logic:

* modResponse is the one switching between the "global states", so modResponse should be the one to switch the states.
* Even though only one state ($STATE_UNCACHED) is needed, adding all 3 states is more orthogonal and also helps quickly determine the state while stepping through the code in a debugger.